pkg/client/unversioned/remotecommand/remotecommand_test.go

```diff
@@ -92,15 +92,21 @@ func fakeExecServer(t *testing.T, i int, stdinData, stdoutData, stderrData, erro
 
 		if len(errorData) > 0 {
 			fmt.Fprint(errorStream, errorData)
+		}
+		if errorStream != nil {
 			errorStream.Close()
 		}
 
 		if len(stdoutData) > 0 {
 			fmt.Fprint(stdoutStream, stdoutData)
+		}
+		if stdoutStream != nil {
 			stdoutStream.Close()
 		}
 		if len(stderrData) > 0 {
 			fmt.Fprint(stderrStream, stderrData)
+		}
+		if stderrStream != nil {
```

Comment:
This test shouldn't have to close anything explicitly... all the streams get Reset() and conn.Close() is called when the function returns. The ioutil.ReadFull() below was what was hanging